runtime/lua/vim/treesitter/languagetree.lua

```diff
@@ -77,14 +77,24 @@ local TSCallbackNames = {
 ---@field private _opts table Options
 ---@field private _parser TSParser Parser for language
 ---@field private _has_regions boolean
----@field private _regions table<integer, Range6[]>?
+---
 ---List of regions this tree should manage and parse. If nil then regions are
 ---taken from _trees. This is mostly a short-lived cache for included_regions()
+---@field private _regions table<integer, Range6[]>?
+---
+---Inverse of `_regions`. start byte of range 1 ↦ { region1, index1, region2, index2, .. }.
```

Comment:
I don't understand this. Can it be explained more?

Reply:
Copied from the comments I added in the latest force push:

Inverse region table, i.e., a (chaining) hash table from regions to their index in `_region`.
Used for checking if an added region is already managed by this parser, so that it can reuse
the existing tree for incremental parsing.
The hash function is simply `region[1][3]` (the start byte of its first range).
Each bucket has the shape of { region1, index of region1, region2, index of region2, ... }.